nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/DataSetBulkAction.java

```diff
@@ -129,26 +130,31 @@ protected void compute(CoreSession coreSession, List<String> ids, Map<String, Se
             List<String> customProperties = asList(split((String) properties.get(EXPORT_FEATURES_PARAM), ","));
             int percentSplit = Integer.parseInt((String) properties.getOrDefault(EXPORT_SPLIT_PARAM, DEFAULT_SPLIT));
             ThreadLocalRandom random = ThreadLocalRandom.current();
-            for (String id : ids) {
-                try {
-                    DocumentModel doc = coreSession.getDocument(new IdRef(id));
-                    BlobTextFromDocument subDoc = PropertyUtils.docSerialize(doc, new HashSet<>(customProperties));
-                    boolean isTraining = random.nextInt(1, 101) <= percentSplit;
-                    if (subDoc != null) {
-                        if (log.isTraceEnabled()) {
-                            log.trace((isTraining ? "training " : "validate ") + subDoc);
-                        }
-                        Record record = toRecord(command.getId(), subDoc);
-                        if (isTraining) {
-                            training.add(record);
-                        } else {
-                            validation.add(record);
-                        }
+            for (DocumentModel doc : loadDocuments(coreSession, ids)) {
+                BlobTextFromDocument subDoc = null;
+                List<String> targetProperties = new ArrayList<>();
+                targetProperties.addAll(customProperties);
```

Comment:
`ArrayList` constructor accepts another collection to create a copy from it.

Reply:
Yes, I started with that, but had a problem.  One of them was unmodifiable so couldn't copy - it was strange.

Reply:
hm, ok. my bad

Reply:
actually, i didn't understand it but changed the code to be sure.